File: test/Models/Buildable/HouseTest.java
package Models.Buildable;

import Models.Measurement;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Vector;

import static org.junit.jupiter.api.Assertions.*;

class HouseTest {
    @Test
    public void shouldAddWall() {
        Vector<Vector<String>> results = new Vector<>();
        results.add(new Vector<>(List.of("Material", "Quantity")));
        results.add(new Vector<>(List.of("24\" 2x4", "3")));
        results.add(new Vector<>(List.of("48\" 2x4", "3")));
        results.add(new Vector<>(List.of("10d nails", "30")));
        assertEquals(results, new House().addWall(new Measurement(24), new Measurement(30)).materials());
    }
}

File: src/test/kotlin/models/buildable/HouseTest.kt
package models.buildable

import models.Measurement
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

internal class HouseTest {
    @Test
    fun shouldAddWall() {
        val results = Vector<Vector<String>>()
        results.add(Vector(listOf("24\" 2x4", "3")))
        results.add(Vector(listOf("48\" 2x4", "3")))
        results.add(Vector(listOf("10d nails", "30")))
        Assertions.assertEquals(results, House().addWall(Measurement(24), Measurement(30)).materials())
    }
}